pmlc/dialect/stdx/transforms/boundscheck.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 Intel Corporation
+
+#include "mlir/ADT/TypeSwitch.h"
+#include "mlir/Conversion/AffineToStandard/AffineToStandard.h"
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Support/DebugStringHelper.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+#include "pmlc/dialect/stdx/transforms/boundscheck.h"
+#include "pmlc/util/logging.h"
+
+using namespace mlir; // NOLINT
+
+namespace pmlc::dialect::stdx {
+
+namespace {
+template <typename LoadStoreOp>
+class BoundsCheckGenerator {
```

Comment:
I would probably not have made a class, but just used two free functions (one templated) for getOrInsertFunc and generateBoundsChecks, but that would only make the code a bit smaller, so probably more of a preference thing.  Or alternately, used an object to store pass-common stuff (probably caching results of generating the function declaration), and then had the op / loc be variables in a member function.